src/fidesops/api/v1/endpoints/privacy_request_endpoints.py

```diff
@@ -154,6 +161,28 @@ def create_privacy_request(
             privacy_request.cache_identity(privacy_request_data.identity)
             privacy_request.cache_encryption(privacy_request_data.encryption_key)
 
+            # Store masking secrets in the cache
+            logger.info(
+                f"Caching masking secrets for privacy request {privacy_request.id}"
+            )
+            erasure_rules: List[Rule] = policy.get_rules_for_action(
+                action_type=ActionType.erasure
+            )
+            unique_masking_strategies_by_name: Set[str] = set()
+            for rule in erasure_rules:
+                unique_masking_strategies_by_name.add(rule.masking_strategy["strategy"])
+            for strategy_name in unique_masking_strategies_by_name:
```

Comment:
This second loop isn't adding any value — you could remove this line entirely and still have the same working code (and just one loop)

Reply:
ended up using an inline set comprehension for brevity: `for strategy_name in {rule.masking_strategy["strategy"] for rule in erasure_rules}:`

Reply:
You could avoid the extra iteration with a check like:
```
if `rule.masking_strategy["strategy"] in unique_masking_strategies_by_name:
    continue
```

Reply:
ahhh nice, makes sense to me!